src/sentry/models/releases/set_commits.py

```diff
@@ -88,6 +94,45 @@ def set_commits_on_release(release, commit_list):
     latest_commit = None
     for idx, data in enumerate(commit_list):
         commit = set_commit(idx, data, release)
+
+        if idx == 0:
+            latest_commit = commit
+
+        commit_author_by_commit[commit.id] = commit.author
+        head_commit_by_repo.setdefault(data["repo_model"].id, commit.id)
+
+    release.update(
+        commit_count=len(commit_list),
+        authors=[
+            str(a_id)
+            for a_id in ReleaseCommit.objects.filter(
+                release=release, commit__author_id__isnull=False
+            )
+            .values_list("commit__author_id", flat=True)
+            .distinct()
+        ],
+        last_commit_id=latest_commit.id if latest_commit else None,
+    )
+    return head_commit_by_repo, commit_author_by_commit
```

Comment:
For anyone looking at this PR after me, the function is `set_commits_on_release` is unchanged regardless of what the diff may seem to indicate.

If `set_commits_on_release_with_transaction_per_commit` was further down in the module the diff would show that `set_commits_on_release` stands untouched.